subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/connection/DeprecatedVersionConsumerConnection.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.internal.consumer.connection;
+
+import org.gradle.api.UncheckedIOException;
+import org.gradle.tooling.BuildAction;
+import org.gradle.tooling.internal.consumer.TestExecutionRequest;
+import org.gradle.tooling.internal.consumer.parameters.ConsumerOperationParameters;
+import org.gradle.tooling.internal.consumer.versioning.VersionDetails;
+import org.gradle.util.GradleVersion;
+
+import java.io.IOException;
+
+public class DeprecatedVersionConsumerConnection implements ConsumerConnection {
+    private static final String DEPRECATION_WARNING = "Support for %s older than %s is deprecated and will be removed in 5.0. You are currently using %s. You should upgrade your %s.\n";
```

Comment:
⭕️ Can you adjust this message to sound more like the one in `UnsupportedOlderVersionConnection`? It provides the user with a little more detail and a more actionable description of what to do about it.

Reply:
Done.